src/lin.rs

```diff
@@ -10,102 +10,69 @@ use BaseDirs;
 use UserDirs;
 use ProjectDirs;
 
-impl BaseDirs {
-    /// Creates a `BaseDirs` struct which holds the paths to user-invisible directories for cache, config, etc. data on the system.
-    /// The returned struct is a snapshot of the state of the system at the time `new()` was invoked.
-    pub fn new() -> BaseDirs {
-        let home_dir       = env::home_dir().unwrap();
-        let cache_dir      = env::var_os("XDG_CACHE_HOME") .and_then(is_absolute_path).unwrap_or_else(|| home_dir.join(".cache"));
-        let config_dir     = env::var_os("XDG_CONFIG_HOME").and_then(is_absolute_path).unwrap_or_else(|| home_dir.join(".config"));
-        let data_dir       = env::var_os("XDG_DATA_HOME")  .and_then(is_absolute_path).unwrap_or_else(|| home_dir.join(".local/share"));
-        let data_local_dir = data_dir.clone();
-        let runtime_dir    = env::var_os("XDG_RUNTIME_DIR").and_then(is_absolute_path);
-        let executable_dir = 
-            env::var_os("XDG_BIN_HOME").and_then(is_absolute_path).unwrap_or_else(|| {
-                let mut new_dir = data_dir.clone(); new_dir.pop(); new_dir.push("bin"); new_dir });
-
-        BaseDirs {
-            home_dir:       home_dir,
-            cache_dir:      cache_dir,
-            config_dir:     config_dir,
-            data_dir:       data_dir,
-            data_local_dir: data_local_dir,
-            executable_dir: Some(executable_dir),
-            runtime_dir:    runtime_dir
-        }
+pub fn base_dirs() -> BaseDirs {
```

Comment:
Is it necessary that these functions are public? I think I tried something like this before and was unhappy that these showed up in the user-facing API of the library.